From 9f2312bc05a8918d00cc8a6223723764133ec511 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 14 Jun 2022 20:36:26 +0300
Subject: [PATCH] [SW-0] common: make progress bits enum device specific

Currently the progress bits enum is common for all platforms. Due to
this, some of the bits remains un-used as those are used by previous
platforms. In order to optimize usage of progress bits, this patch
adds device specific progress bits enum.

Change-Id: I8e48dfc43fe1ded41a234224d8cb9c05b23f6f95
---

diff --git a/arch/gaudi2/arc/progress_bits.h b/arch/gaudi2/arc/progress_bits.h
new file mode 100644
index 0000000..c4c7d63
--- /dev/null
+++ b/arch/gaudi2/arc/progress_bits.h
@@ -0,0 +1,111 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+enum arcboot_rom_progress {
+	PROGRESS_INIT = 0,
+	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
+	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
+	PROGRESS_REVOKE_LOC = 15,
+	PROGRESS_REVOKE_LOC_0 = 16,
+	PROGRESS_REVOKE_LOC_1 = 17,
+	PROGRESS_KEY_LOC = 18,
+	PROGRESS_KEY_LOC_0 = 19,
+	PROGRESS_KEY_LOC_1 = 20,
+	PROGRESS_VER_LOC = 21,
+	PROGRESS_VER_LOC_0 = 22,
+	PROGRESS_VER_LOC_1 = 23,
+	PROGRESS_BRINGUP_PCI_PRI = 24,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 25,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 26,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 27,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 28,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 29,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 30,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 31,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 32,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 33,
+	PROGRESS_BRINGUP_PCI_SEC = 34,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 35,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 36,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 37,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 38,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 39,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 40,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 41,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 42,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 43,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 44,
+	PROGRESS_PPBOOT_START = 45,
+	PROGRESS_PPBOOT_FINISH = 46,
+	PROGRESS_BRINGUP_RAM_PRI = 47,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 48,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 49,
+	PROGRESS_BRINGUP_RAM_SEC = 50,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 51,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 52,
+	PROGRESS_ARCBOOT_RAM_START = 53,
+	PROGRESS_ARCBOOT_RECOVERY = 54,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 55,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 56,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 57,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 58,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 59,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 60,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 61,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 62,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 63,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 64,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 65,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 66,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 67,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 68,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 69,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 70,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 71,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 72,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 73,
+	PROGRESS_ARCBOOT_EROM_PCIE = 74,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 75,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 76,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 77,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 78,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 79,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 80,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 81,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 82,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 83,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 84,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 85,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 86,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 87,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 88,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 89,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 90,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 91,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 92,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 93,
+	PROGRESS_ARCBOOT_EROM_TPM = 94,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 95,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 96,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 97,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 98,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 99,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 100,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 101,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 102,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 103,
+};
diff --git a/arch/gaudi3/arc/progress_bits.h b/arch/gaudi3/arc/progress_bits.h
new file mode 100644
index 0000000..b486457
--- /dev/null
+++ b/arch/gaudi3/arc/progress_bits.h
@@ -0,0 +1,136 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+enum arcboot_rom_progress {
+	PROGRESS_INIT = 0,
+	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
+	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
+	PROGRESS_INIT_SPI_BOOT_DESC_FAIL = 15,
+	PROGRESS_INIT_EFUSE_SPI = 16,
+	PROGRESS_REVOKE_LOC = 17,
+	PROGRESS_REVOKE_LOC_0 = 18,
+	PROGRESS_REVOKE_LOC_1 = 19,
+	PROGRESS_KEY_LOC = 20,
+	PROGRESS_KEY_LOC_0 = 21,
+	PROGRESS_KEY_LOC_1 = 22,
+	PROGRESS_VER_LOC = 23,
+	PROGRESS_VER_LOC_0 = 24,
+	PROGRESS_VER_LOC_1 = 25,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 26,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 27,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 28,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 29,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 30,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 31,
+	PROGRESS_BRINGUP_PCI_PRI = 32,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 33,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 34,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 35,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 36,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 37,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 38,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 39,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 40,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 41,
+	PROGRESS_BRINGUP_PCI_SEC = 42,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 43,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 44,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 45,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 46,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 47,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 48,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 49,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 50,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 51,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 52,
+	PROGRESS_ARCBOOT_ROM_TPM = 53,
+	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 54,
+	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 55,
+	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 56,
+	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 57,
+	PROGRESS_BRINGUP_TPM_SPI = 58,
+	PROGRESS_BRINGUP_TPM_SPI_PRI = 59,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 60,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 61,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 62,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 63,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 64,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 65,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 66,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 67,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 68,
+	PROGRESS_BRINGUP_TPM_SPI_SEC = 69,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 70,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 71,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 72,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 73,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 74,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 75,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 76,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 77,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 78,
+	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 79,
+	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 80,
+	PROGRESS_PPBOOT_START = 81,
+	PROGRESS_PPBOOT_FINISH = 82,
+	PROGRESS_BRINGUP_RAM_PRI = 83,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 84,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 85,
+	PROGRESS_BRINGUP_RAM_SEC = 86,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 87,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 88,
+	PROGRESS_ARCBOOT_RAM_START = 89,
+	PROGRESS_ARCBOOT_RECOVERY = 90,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 91,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 92,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 93,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 94,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 95,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 96,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 97,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 98,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 99,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 100,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 101,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 102,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 103,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 104,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 105,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 106,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 107,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 108,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 109,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 110,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 111,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 112,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 113,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 114,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 115,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 116,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 117,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 118,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 119,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 120,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 121,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 122,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 123,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 124,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 125,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 126,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 127,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 128,
+};
diff --git a/arch/greco/arc_elf32/progress_bits.h b/arch/greco/arc_elf32/progress_bits.h
new file mode 100644
index 0000000..c4c7d63
--- /dev/null
+++ b/arch/greco/arc_elf32/progress_bits.h
@@ -0,0 +1,111 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+enum arcboot_rom_progress {
+	PROGRESS_INIT = 0,
+	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
+	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
+	PROGRESS_REVOKE_LOC = 15,
+	PROGRESS_REVOKE_LOC_0 = 16,
+	PROGRESS_REVOKE_LOC_1 = 17,
+	PROGRESS_KEY_LOC = 18,
+	PROGRESS_KEY_LOC_0 = 19,
+	PROGRESS_KEY_LOC_1 = 20,
+	PROGRESS_VER_LOC = 21,
+	PROGRESS_VER_LOC_0 = 22,
+	PROGRESS_VER_LOC_1 = 23,
+	PROGRESS_BRINGUP_PCI_PRI = 24,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 25,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 26,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 27,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 28,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 29,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 30,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 31,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 32,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 33,
+	PROGRESS_BRINGUP_PCI_SEC = 34,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 35,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 36,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 37,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 38,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 39,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 40,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 41,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 42,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 43,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 44,
+	PROGRESS_PPBOOT_START = 45,
+	PROGRESS_PPBOOT_FINISH = 46,
+	PROGRESS_BRINGUP_RAM_PRI = 47,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 48,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 49,
+	PROGRESS_BRINGUP_RAM_SEC = 50,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 51,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 52,
+	PROGRESS_ARCBOOT_RAM_START = 53,
+	PROGRESS_ARCBOOT_RECOVERY = 54,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 55,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 56,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 57,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 58,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 59,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 60,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 61,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 62,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 63,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 64,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 65,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 66,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 67,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 68,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 69,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 70,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 71,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 72,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 73,
+	PROGRESS_ARCBOOT_EROM_PCIE = 74,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 75,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 76,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 77,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 78,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 79,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 80,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 81,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 82,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 83,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 84,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 85,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 86,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 87,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 88,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 89,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 90,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 91,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 92,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 93,
+	PROGRESS_ARCBOOT_EROM_TPM = 94,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 95,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 96,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 97,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 98,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 99,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 100,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 101,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 102,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 103,
+};
diff --git a/include/arcboot.h b/include/arcboot.h
index 2b1f47d..797a5e8 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -11,6 +11,7 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <stdbool.h>
+#include <progress_bits.h>
 #include "soc.h"
 #include "regmap.h"
 #include "regmask.h"
@@ -136,168 +137,6 @@
 	PROGRESS_EROM_IMG_VERIFICATION_FAIL,
 	PROGRESS_EROM_IMG_START,
 	PROGRESS_EROM_IMG_FINISH,
-};
-
-enum arcboot_rom_progress {
-	PROGRESS_INIT = 0,
-	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
-	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
-	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
-	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
-	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
-	PROGRESS_INIT_SPI_BOOT_DESC_FAIL = 15,
-	PROGRESS_INIT_EFUSE_SPI = 16,
-	PROGRESS_REVOKE_LOC = 17,
-	PROGRESS_REVOKE_LOC_0 = 18,
-	PROGRESS_REVOKE_LOC_1 = 19,
-	PROGRESS_KEY_LOC = 20,
-	PROGRESS_KEY_LOC_0 = 21,
-	PROGRESS_KEY_LOC_1 = 22,
-	PROGRESS_VER_LOC = 23,
-	PROGRESS_VER_LOC_0 = 24,
-	PROGRESS_VER_LOC_1 = 25,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 26,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 27,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 28,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 29,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 30,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 31,
-	PROGRESS_BRINGUP_PCI_PRI = 32,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 33,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 34,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 35,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 36,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 37,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 38,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 39,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 40,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 41,
-	PROGRESS_BRINGUP_PCI_SEC = 42,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 43,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 44,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 45,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 46,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 47,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 48,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 49,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 50,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 51,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 52,
-	PROGRESS_ARCBOOT_ROM_TPM = 53,
-	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 54,
-	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 55,
-	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 56,
-	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 57,
-	PROGRESS_BRINGUP_TPM_SPI = 58,
-	PROGRESS_BRINGUP_TPM_SPI_PRI = 59,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 60,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 61,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 62,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 63,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 64,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 65,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 66,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 67,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 68,
-	PROGRESS_BRINGUP_TPM_SPI_SEC = 69,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 70,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 71,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 72,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 73,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 74,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 75,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 76,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 77,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 78,
-	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 79,
-	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 80,
-	PROGRESS_PPBOOT_START = 81,
-	PROGRESS_PPBOOT_FINISH = 82,
-	PROGRESS_BRINGUP_RAM_PRI = 83,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 84,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 85,
-	PROGRESS_BRINGUP_RAM_SEC = 86,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 87,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 88,
-	PROGRESS_ARCBOOT_RAM_START = 89,
-	PROGRESS_ARCBOOT_RECOVERY = 90,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 91,
-	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 92,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 93,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 94,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 95,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 96,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 97,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 98,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 99,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 100,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 101,
-	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 102,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 103,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 104,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 105,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 106,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 107,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 108,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 109,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 110,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 111,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 112,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 113,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 114,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 115,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 116,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 117,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 118,
-	PROGRESS_ARCBOOT_EROM_PCIE = 119,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 120,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 121,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 122,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 123,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 124,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 125,
-	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 126,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 127,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 128,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 129,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 130,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 131,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 132,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 133,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 134,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 135,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 136,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 137,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 138,
-	PROGRESS_ARCBOOT_EROM_TPM = 139,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 140,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 141,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 142,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 143,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 144,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 145,
-	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 146,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 147,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 148,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 149,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 150,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 151,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 152,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 153,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 154,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 155,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 156,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 157,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 158,
 };
 
 enum security_location {
